OMCompiler/Compiler/NFFrontEnd/NFSimplifyExp.mo

```diff
@@ -1594,7 +1594,7 @@ algorithm
     then result;
 
     // case 2.0 binary SOME(op) inverse operator (-, :)
-    case (SOME(op), Expression.BINARY()) guard(Operator.isCombineable(op, exp.operator)) algorithm
+    case (SOME(op), Expression.BINARY()) guard(Operator.isCombineable(op, exp.operator) and not Operator.isScalarProduct(exp.operator)) algorithm
```

Comment:
Why not change the definition of `isCombinable`?

Reply:
yes i could do that, neither of the operators should be scalar product i guess